jar-connector/src/main/java/com/sixsq/slipstream/connector/UsageRecorder.java

```diff
@@ -75,9 +75,8 @@ public static void insertEnd(String instanceId, String user, String cloud, List<
 			}
 
 			if(!hasRecorded(cloud, instanceId)){
-				logger.fine("Not recorded => avoiding inserting usage record END for "
+				logger.fine("Not recorded but still inserting usage record END for "
 						+ metrics + ", " + describe(instanceId, user, cloud));
-				return;
```

Comment:
What I meant was more related to the metrics processor when it sees `end` w/o `start` (in DB).  Which can be a result of `still inserting end` even if we there is no `cloud, instanceId` in the `recordedVmInstanceIds` (`hasRecoreded`), but not due to the service restart and loss of the records in `recordedVmInstanceIds` map, but rather real absence of the `start`.